includes/wc-formatting-functions.php

```diff
@@ -271,11 +271,11 @@ function wc_format_decimal( $number, $dp = false, $trim_zeros = false ) {
 
 	// DP is false - don't use number format, just return a string in our format
 	} elseif ( is_float( $number ) ) {
-		$number = wc_clean( str_replace( $decimals, '.', strval( $number ) ) );
+		$number = strval( $number );
 	}
 
 	if ( $trim_zeros && strstr( $number, '.' ) ) {
-		$number = rtrim( rtrim( $number, '0' ), '.' );
+		$number = rtrim( trim( $number, '0' ), '.' );
```

Comment:
Think about numbers like "018.50". With trim_zeros this should return "18.5", not "018.5"